### DISCUSSION TITLE
Error installing package

### DISCUSSION THREAD

**Discussion by te0203** [jaredhendrickson13/pfsense-api#593] [General]
Good day. 
After installing the package, an error appeared:

Crash report begins.  Anonymous machine information:

amd64
14.0-CURRENT
FreeBSD 14.0-CURRENT amd64 1400094 #1 RELENG_2_7_2-n255948-8d2b56da39c: Wed Dec  6 20:45:47 UTC 2023     root@freebsd:/var/jenkins/workspace/pfSense-CE-snapshots-2_7_2-main/obj/amd64/StdASW5b/var/jenkins/workspace/pfSense-CE-snapshots-2_7_2-main/sources/F

Crash report details:

PHP Errors:
[03-Nov-2024 18:02:56 Europe/Kyiv] PHP Fatal error:  Uncaught RESTAPI\Responses\ServerError: Cannot locate first Model object in a ModelSet with no Model objects in /usr/local/pkg/RESTAPI/Core/ModelSet.inc:68
Stack trace:
#0 /usr/local/pkg/RESTAPI/Models/RoutingGatewayGroup.inc(71): RESTAPI\Core\ModelSet->first()
#1 /usr/local/pkg/RESTAPI/Core/Model.inc(885): RESTAPI\Models\RoutingGatewayGroup->from_internal_ipprotocol('unknown')
#2 /usr/local/pkg/RESTAPI/Core/Model.inc(1788): RESTAPI\Core\Model->from_internal_object(Array)
#3 /usr/local/pkg/RESTAPI/Fields/ForeignModelField.inc(310): RESTAPI\Core\Model::read_all()
#4 /usr/local/pkg/RESTAPI/Fields/ForeignModelField.inc(329): RESTAPI\Fields\ForeignModelField->get_in_scope_models()
#5 /usr/local/pkg/RESTAPI/Fields/ForeignModelField.inc(226): RESTAPI\Fields\ForeignModelField->__get_matches('name', 'WANGW')
#6 /usr/local/pkg/RESTAPI/Core/Field.inc(905): RESTAPI\Fields\ForeignModelField->_from_internal('WANGW')
#7 /usr/local/pkg/RESTAPI/Core/Model.inc(858): RESTAPI\Core\Field->from_internal('WANGW')
#8 /usr/local/pkg/RESTAPI/Core/Model.inc(1788): RESTAPI\Core\Model->from_internal_object(Array)
#9 /usr/local/pkg/RESTAPI/Core/Model.inc(1828): RESTAPI\Core\Model::read_all(NULL, 0, 0, false)
#10 /usr/local/pkg/RESTAPI/Fields/InterfaceField.inc(189): RESTAPI\Core\Model::query(Array)
#11 /usr/local/pkg/RESTAPI/Core/Field.inc(323): RESTAPI\Fields\InterfaceField->get_interface_choices()
#12 /usr/local/pkg/RESTAPI/Core/Field.inc(1125): RESTAPI\Core\Field->set_choices_from_callable()
#13 /usr/local/pkg/RESTAPI/Core/Form.inc(195): RESTAPI\Core\Field->to_form_input()
#14 /usr/local/pkg/RESTAPI/Core/Form.inc(354): RESTAPI\Core\Form->get_form()
#15 /usr/local/www/system_restapi_settings.php(4): RESTAPI\Core\Form->print_form()
#16 {main}
  thrown in /usr/local/pkg/RESTAPI/Core/ModelSet.inc on line 68
[03-Nov-2024 18:03:03 Europe/Kyiv] PHP Fatal error:  Uncaught RESTAPI\Responses\ServerError: Cannot locate first Model object in a ModelSet with no Model objects in /usr/local/pkg/RESTAPI/Core/ModelSet.inc:68

Current Versionv 2.1.3
Bookmark REST API->settings not working
Please help me.

**Comment by jaredhendrickson13** [comment]
Can you provide the `gateways` section of your XML config? The error suggests an interface is using a gateway group that references a gateway that does not currently exist in config.

**Comment by te0203** [comment]
[gateways.xml.txt](https://github.com/user-attachments/files/17612014/gateways.xml.txt)
